### PR TITLE
[FSSS-199] Implement/style Search results page

### PR DESCRIPTION
**Pull request by heitorado**
## What's the purpose of this pull request?
This PR styles the search page results. Below are the images for reference.

- Desktop
Screen Shot 2022-03-16 at 17 36 03
Screen Shot 2022-03-16 at 17 36 30

- Mobile
Screen Shot 2022-03-16 at 17 37 24
Screen Shot 2022-03-16 at 17 37 41

## How does it work?
Nothing unusual, just some extra props being passed and changes/additions in HTML/CSS here and there.

## How to test it?
- `yarn test`, verify that all tests still pass
- Run the development server
  - Search something that exists in the store (I usually use the word `tasty`)
    - Verify that the search page results show up and it is styled like the images above, both on mobile and desktop.
  - Search something that does not exist in the store (I usually use the phrase `brastemp washing machine`) 
    - Verify that the empty gallery (no results found) page renders and it is styled like the images above, both on mobile and desktop

## References
Thanks to @danzanzini for all the help

## Checklist
- [x] CHANGELOG entry added

### CODE REVIEW

File: src/components/sections/ProductGallery/product-gallery.scss [hunk 1]
```diff
@@ -4,12 +4,25 @@
...
   @include media("<notebook") {
     padding-top: 0;
   }
 }
 
+.product-listing__search-term {
+  margin-bottom: var(--space-3);
+
+  h1 {
+    font-size: var(--text-size-5);
```
Comment (eduardoformiga) [inline src/components/sections/ProductGallery/product-gallery.scss:18]: I think on mobile (mobile first) we should use the `--text-size-4` (25px)

```suggestion
    font-size: var(--text-size-4);

    @include media(">=tablet") { font-size: var(--text-size-5); }
```

**Review by eduardoformiga — APPROVED** [review]
Good Job! 👏 

I left some minor comments.

File: src/pages/s.tsx [hunk 2]
```diff
@@ -54,13 +55,14 @@ function Page(props: Props) {
         }}
       />
 
-      {/*
-        Sections: Components imported from '../components/sections' only.
-        Do not import or render components from any other folder in here.
-      */}
```
[the remark this answers is not in the document]
Comment (heitorado) [reply src/pages/s.tsx:60]: I agree, I just do not think this was the right place for it. Maybe we should leave this only inside the `Section` component?

File: src/components/sections/ProductGallery/product-gallery.scss [hunk 3]
```diff
@@ -4,12 +4,27 @@
   --product-listing-row-height: var(--space-6);
 
   height: 100%;
+  padding-top: 0;
```
Comment (filipewl) [inline src/components/sections/ProductGallery/product-gallery.scss:7]: This seems to be needed for the search page, but the existing padding is needed for PLP. 
😕 

https://user-images.githubusercontent.com/381395/158908807-66f4f054-4c76-4670-9730-c5a05529cb25.mp4

CleanShot 2022-03-17 at 20 09 31

Now how to solve this only for the search page... 
🤔

File: src/pages/s.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (filipewl) [reply src/pages/s.tsx:60]: I guess this comment didn't stop us from not following it when we were making our changes. But we should've paid more attention.
😆 

I like having more details about why this is needed on the documentation somewhere.

And if we're going to remove it, we should remove them altogether (there are 2 other occurrences in the Home and PDP pages).

**Review by filipewl — COMMENTED** [review]
Looking very good! 👏

**Comment by filipewl** [comment]
I wonder if we should include the search page to be run on LH to make sure it's consistent with the rest?

**Comment by filipewl** [comment]
Does this task also encompass this option for clearing the search term?

CleanShot 2022-03-17 at 20 25 19

**Comment by heitorado** [comment]
> Does this task also encompass this option for clearing the search term?
> 
I've aligned with the designers and we won't be doing that, since it will be removed soon on the next version.

File: src/pages/s.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (heitorado) [reply src/pages/s.tsx:60]: This was making sense to me some commits ago, where the `Section` component was being used explicitly in that file as a wrapper for other components, so the warning was coherent: It should warn us that not everything could be put inside the `Section` component, just the ones inside `components/sections` (I think).

However, some commits later I noticed that it vanished from that file (it was moved), and now the `Section` component resides inside each component it wraps (look at `Breadcrumb.tsx` for an example) - it is not being called "from outside" anymore. But the comment remains.

File: src/components/sections/ProductGallery/product-gallery.scss (hunk 3 above)
Comment (heitorado) [reply src/components/sections/ProductGallery/product-gallery.scss:7]: After aligning with the designers we changed the design a bit to accomodate the title gracefully, not interfering with the PLP style.

Comment (heitorado) [reply src/components/sections/ProductGallery/product-gallery.scss:7]: After aligning with the designers we changed the design a bit to accommodate the title gracefully, not interfering with the PLP style.

File: src/pages/s.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (heitorado) [reply src/pages/s.tsx:60]: @filipewl more about sections (and the `Section` component) here: https://github.com/vtex-sites/base.store/pull/371

TL;DR:
> The whole idea of sections is to split app logic into replaceable/reusable strips of screen content so that CMS users can easily replace/rearrange the desired component on the screen. (...)
> This means that the Section map has to render a self contained section. 
> (...) so CMS users can correctly change and re-order the sections

[the remark this answers is not in the document]
Comment (heitorado) [reply src/pages/s.tsx:60]: I'll rewrite and bring back the comments, now that their purpose is clearer.

**Review by filipewl — APPROVED** [review]
Looks good to me!

**Review by ArthurTriis1 — APPROVED** [review]
